### PR TITLE
Fix a potential bug for android demo

### PR DESCRIPTION
**Pull request by philokey**
Size of the priority queue will decrease in every iteration, so the terminal condition will change,  but i increase in every iteration, it will lead to some problems. For example, when `MAX_RESULTS = 10 `and` pq.size() = 4` in the beginning, after 2 iterations,  `i = 2`, the  terminal condition will become `i < 2`, but it  is wrong.

### CODE REVIEW

**Comment by vrv** [comment]
Do you plan on signing the CLA?

File: tensorflow/examples/android/src/org/tensorflow/demo/TensorFlowImageClassifier.java [hunk 1]
```diff
@@ -160,7 +160,8 @@ public int compare(Recognition lhs, Recognition rhs) {
       }
     }
     final ArrayList<Recognition> recognitions = new ArrayList<Recognition>();
-    for (int i = 0; i < Math.min(pq.size(), MAX_RESULTS); ++i) {
+    int recognitions_size = Math.min(pq.size(), MAX_RESULTS);
+    for (int i = 0; i < recognitions_size; ++i) {
```
Comment (philokey) [inline tensorflow/examples/android/src/org/tensorflow/demo/TensorFlowImageClassifier.java]: I fixed the problem.

**Comment by philokey** [comment]
@vrv Yes, The application has just been submitted。

**Comment by andrewharp** [comment]
Jenkins, test this please.

**Comment by andrewharp** [comment]
Thanks for the fix!

**Pull request merged**